android/app/src/main/java/com/created/team201/presentation/common/customview/dayofselector/DayOfWeekSelector.kt

```diff
@@ -0,0 +1,167 @@
+
+        if (canMultiSelect) {
+            dayTextViews.values.forEach {
+                it.background = ResourcesCompat.getDrawable(
+                    resources,
+                    R.drawable.bg_day_of_week_selector_multi_select,
+                    null,
+                )
+            }
+        } else {
+            dayTextViews.values.forEach {
+                it.background = ResourcesCompat.getDrawable(
+                    resources,
+                    R.drawable.bg_day_of_week_selector_single_select,
+                    null,
+                )
+            }
+        }
+
+        typedArray.recycle()
+    }
+
+    private fun setupDayClickListeners() {
+        dayTextViews.values.forEach { textView ->
+            textView.setOnClickListener { it ->
+                val clickedDay: DayOfWeek = textViewDays[it] ?: throw NullPointerException()
+                if (canMultiSelect) {
+                    selectDayTextView(textView)
+                    dayOnClick?.onClick(clickedDay)
+                    return@setOnClickListener
+                }
+
+                if (it.isSelected) return@setOnClickListener
+
+                selectDayTextView(textView)
+                dayTextViews.values
+                    .filterNot { it == textView }
+                    .forEach { it.isSelected = false }
+                dayOnClick?.onClick(clickedDay)
+            }
+        }
+    }
+
+    private fun setupDayTexts() {
+        val days: List<DayOfWeek> = DayOfWeek.getValuesWithStartDay()
+        dayTextViews.values.forEachIndexed { index, textView ->
+            textView.text = resources.getString(days[index].stringRes)
+        }
+    }
+
+    fun setSelectableDays(selectableDays: List<DayOfWeek>) {
+        dayTextViews.keys.forEach {
+            if (selectableDays.contains(it)) {
+                dayTextViews[it]?.isEnabled = true
+                return@forEach
+            }
+            dayTextViews[it]?.isEnabled = false
+        }
+    }
+
+    fun selectDay(day: DayOfWeek) {
+        dayTextViews[day]?.isSelected = true
+    }
+
+    fun selectDays(days: List<DayOfWeek>) {
+        days.forEach { day ->
+            val textView: TextView = dayTextViews[day]
+                ?: throw NullPointerException("$day 에 해당하는 TextView가 존재하지 않습니다.")
+
+            textView.isSelected = !(textView.isSelected)
+        }
+    }
+
+    private fun selectDayTextView(day: TextView) {
+        day.isSelected = !day.isSelected
+    }
+
+    fun getSelectedDays(): List<DayOfWeek> {
+        val selectedDayTextViews: List<TextView> = dayTextViews.values.filter { it.isSelected }
+        return selectedDayTextViews.map { textViewDays[it] ?: throw NullPointerException() }
+    }
+
+    fun setDayOnClickListener(dayOnClickListener: DayOnClickListener) {
+        dayOnClick = dayOnClickListener
+    }
+
+    fun getSelectedDaysSize(): Int {
+        return dayTextViews.values.filter { it.isSelected }.size
```

Comment:
아래 코드로 변경해도 될 거 같은데 아닌가염? (아님 망고~)
```suggestion
        return dayTextViews.values.count { it.isSelected }
```

Reply:
이건 ㄹㅇ 바보 오브 레전드네요 수정하였습니다 ㅋㅋㅋ